cli/src/error.rs

```diff
@@ -0,0 +1,92 @@
+//! Error types
```

Comment:
The CLI previously relied on `failure::Error`, which was kind of a hack. This file is in-line with what's in the current boilerplate for an error type.